saleor/product/tasks.py

```diff
@@ -35,15 +37,23 @@ def _update_variants_names(instance: ProductType, saved_attributes: Iterable):
     attributes_changed = initial_attributes.intersection(saved_attributes)
     if not attributes_changed:
         return
-    variants_to_be_updated = ProductVariant.objects.filter(
+
+    variants = ProductVariant.objects.filter(
         product__in=instance.products.all(),
         product__product_type__variant_attributes__in=attributes_changed,
     )
-    variants_to_be_updated = variants_to_be_updated.prefetch_related(
-        "attributes__values__translations"
-    ).all()
-    for variant in variants_to_be_updated:
-        generate_and_set_variant_name(variant, variant.sku)
+
+    variants_to_update = []
+    for variant in variants:
+        variants_to_update.append(
+            generate_and_set_variant_name(variant, variant.sku, save=False)
+        )
+        if len(variants_to_update) > VARIANTS_UPDATE_BATCH:
```

Comment:
This should be fine but couldn't we reuse `queryset_in_batches` pattern from migrations, just to keep this kind of operations unified?

Reply:
@maarcingebala 
Not preferably since `queryset_in_batches` operates on `pks` (yields `batch_pks`), then `bulk_update` accepts an iterable of Django objects it would require an extra SQL query for each batch to turn pks into such.

Reply:
👍